aleph/views/bookmarks_api.py

```diff
@@ -123,3 +124,57 @@ def delete(entity_id):
     db.session.commit()
 
     return "", 204
+
+
+@blueprint.route("/api/2/bookmarks/migrate", methods=["POST"])
+def migrate():
+    """Migrate bookmarks.
+    ---
+    post:
+      summary: Migrate bookmarks
+      tags: [Bookmarks]
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/BookmarkMigrate'
+      responses:
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  errors:
+                    items:
+                      type: string
+                      format: entity-id
+    """
+    require(request.authz.session_write)
+    data = parse_request("BookmarkMigrate")
+    values = []
+    errors = []
+
+    for bookmark in data:
+        try:
+            entity = get_index_entity(bookmark.get("entity_id"))
```

Comment:
Could instead use `entities_by_ids` from `aleph.index.entities` to get all entities in a single ES request, but would need to manually check permissions.

Reply:
Is this following similar conventions in other parts of the codebase? What I'm saying is that if other areas are manually checking the permissions it might be worth following the convention, if not, then that's a good indicator that perhaps this is the way to go

Reply:
Yes and no. For singular resources, we do rely on these helpers to check permissions and throw appropriate errors in case of missing permissions etc.

For lists of many resources, we usually ensure that only resources the current user can access are returned at the query level, and we have another last-resort check when serializing resources in API responses.

This case is a little different though -- we’re not processing data from the database, but from a user. And we can’t rely on the serializer-level check because we actually need to know about the issues in order to show a message to the user.

The current solution is the least complex, but I’m not sure if it might lead to issues when migrating huge lists of bookmarks due to the O(n) ES requests. So far it hasn’t been a problem locally, and I’ll make sure to test it on staging once it’s out there.